### DISCUSSION TITLE
React - selected value is not displayed after refreshing page

### DISCUSSION THREAD

**Discussion by abdulmateentechbits** [react-hook-form/react-hook-form#9228] [Q&A]
Please help! I have my re usable SelectField Component when i refresh the page after page load initially the selected value is not display is there any solution in react hook form to persist that value even page is refresh
Thankyou !
@bluebill1049 
@leapful 
@NMinhNguyen

**Comment by Moshyfawn** [comment]
React state is not persisted between page refreshes.

You want to store your form's values / select field value in some external storage like local storage / URL / server and populate `useForm` `defaultValues` with it
